### DISCUSSION TITLE
Questions and clarifications about using publiccode.yml

### DISCUSSION THREAD

**Discussion by dergyitheron** [publiccodeyml/publiccode.yml#138] [Q&A]
Hello everyone,

I am coming to you with some questions and things I would like to have clarified from position of a SW developer. I find your description of this project not entirely clear in terms why would I want to use publiccode.yml in my open source projects or if I even should. 

I see a lot of description aimed at governments and software catalogues but not many are explaining actual developers why would this be beneficial for them.

My questions:
1. I need some standardized way of adding metadata to my OSS repositories that would show version, name, description, maintainer, some categories etc. This standard fits this perfectly, can I simply use it for this?
2. Should I care about national software catalogues? I personally don't know what benefits would it have to have my software in some local catalogue maintained by government.
3. Speaking of catalogues, does it make sense to use publiccode.yml and generate my own catalogue of my own projects? How you, maintainers, feel about this?
4. If the answer to 3. is somewhat yes, could I use publiccode.yml in my company to create catalogue just for internal documentation purposes? 

Last two questions are mostly there because I miss some standard like this in general. I know this is meant to be used with public software but I am finding it hard to understand your licensing therefore I'm asking.

Thinking about it now, FAQ section with such questions would be really nice. A lot of your terms are confusing to me and I find it hard to see the exact definition of these, such as free software developer, public software vendor, public servant, civil servants, public administrations, reuse of projects. Haven't heard many of these terms in context of OSS until today.q

Thank you very much.

David

**Comment by bfabio** [comment]
Hello @dergyitheron,

you bring up some really great points.

Right now the only descriptions of the project are the README from the GitHub repo (which has a good margin of improvement) and the Standard documentation at [https://yml.publiccode.tools](https://yml.publiccode.tools), which is a technical document with little context.

The good news is that we should soon be able to work on a simple site with FAQs and a user friendly description of the project, possibly from the point of view of different personas. It'll probably be at [https://publiccode.tools](https://publiccode.tools) (right now a blank page). Your questions are totally FAQ material, thanks for that :)

> I need some standardized way of adding metadata to my OSS repositories that would show version, name, description, maintainer, some categories etc. This standard fits this perfectly, can I simply use it for this?

Yes! The Standard itself is [CC0 1.0](https://github.com/publiccodeyml/publiccode.yml/blob/main/LICENSE) licensed.

You can also use these tools:
* **https://publiccode-editor.developers.italia.it/** ([GitHub repo](https://github.com/italia/publiccode-editor)), a visual editor to generate `publiccode.yml` files - some fields are Italy centric, but it can be useful nonetheless
* **https://github.com/italia/publiccode-parser-go**, the reference implementation parser and validator
* **https://github.com/italia/publiccode-softwareversion-check-action**, a GitHub action to keep versions in `publiccode.yml` in sync with tags in the repo.

[more tools...](https://github.com/italia?q=publiccode&type=all&language=&sort=)

> Should I care about national software catalogues? I personally don't know what benefits would it have to have my software in some local catalogue maintained by government.

Not really, if you don't want to. One benefit is potentially having more visibility with a specific audience (typically [public employees](https://en.wikipedia.org/wiki/Civil_service)). [Example of the Italian one](https://developers.italia.it/en/search?type=all_catalogue&sort_by=release_date&page=0#bd5f4201143feece27775675aa056949db09f6d1)

> Speaking of catalogues, does it make sense to use publiccode.yml and generate my own catalogue of my own projects? How you, maintainers, feel about this?
> If the answer to 3. is somewhat yes, could I use publiccode.yml in my company to create catalogue just for internal documentation purposes?

I'd say go for it! The Standard was originally created as the metadata format for catalogs, but at the end of the day it's just a file format and can be used for other ends as well.

> A lot of your terms are confusing to me and I find it hard to see the exact definition of these

I'd be great if we could go into detail here, term by term, so we can improve the documentation, having a fresh take is invaluable! We can either do it here, or with PRs ([CONTRIBUTING.md](https://github.com/publiccodeyml/publiccode.yml/blob/main/CONTRIBUTING.md), [procedure-proposing-changes-and-voting.md](https://github.com/publiccodeyml/publiccode.yml/blob/main/governance/procedure-proposing-changes-and-voting.md))
